### PR TITLE
extract domain from cli genspider URL

### PR DESCRIPTION
**Pull request by matsievskiysv**
This PR modifies behavior of cli `genspider` command to allow passing not only domain names, but also URLs. 
For example, `https://test.com/`, `http://test.com/`, `https://test.com/other` and `test.com/` arguments would all produce spider with `allowed_domains = ['test.com']` and `start_urls = ['http://test.com/']`

This is particularly useful for zsh users because autocompletion prepends schema to domain argument.

### CODE REVIEW

**Comment by kmike** [comment]
I think that's a great idea. 

Could you please add tests?

**Comment by seregaxvm** [comment]
@kmike added some tests

**Comment by seregaxvm** [comment]
Sorry, didn't notice failed tests until now.

File: tests/test_commands.py [hunk 1]
```diff
@@ -79,6 +80,15 @@ def kill_proc():
 
         return p, to_unicode(stdout), to_unicode(stderr)
 
+    def find_in_file(self, filename, regex):
+        '''Find first pattern occurrence in file'''
```
Comment (Gallaecio) [inline tests/test_commands.py]:
```suggestion
        """Find first pattern occurrence in file"""
```

:lipstick: I think this is the only area were PEP8 actually makes a choice between `'` and `"` :slightly_smiling_face:

File: tests/test_commands.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import subprocess
 import sys
 import tempfile
+import re
```
Comment (Gallaecio) [inline tests/test_commands.py]: :lipstick: Let’s keep the imports sorted alphabetically.

```suggestion
import re
import subprocess
import sys
import tempfile
```

**Review by wRAR — APPROVED** [review]

**Review by Gallaecio — APPROVED** [review]

**Comment by Gallaecio** [comment]
Great, thanks!

**Comment by wRAR** [comment]
Should we now close various other issues/PRs about passing an URL to genspider?

**Pull request merged**